### PR TITLE
feat(ios):  update Ti.UI.View.borderRadius to accept number, array and string

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-27305
Note - Merge with PR #11796 , which have doc changes. Unit test is copied from there.

### CODE REVIEW

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m [hunk 1]
```diff
@@ -644,19 +647,82 @@ - (void)setBackgroundTopCap_:(id)value
...
+    bottomLeftRadius = radius;
+    topRightRadius = radius;
+  } else if (radiusArray.count == 1) {
+    // For same corner radius, no need to create bezier path. Use CALayer's cornerRadius.
+    viewLayer.cornerRadius = [self radiusFromObject:radiusArray[0]];
+    return;
+  }
+
+  CAShapeLayer *shapeLayer = [[CAShapeLayer alloc] initWithLayer:viewLayer];
+  CGRect rect = viewLayer.bounds;
```
Comment (janvennemann) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m:683]: I don't see any other usage of `rect` apart from `rect.size`, so why not directly pull the size and use only that? Avoids having to type `rect.size.*` again and again.

```suggestion
  CGSize size = viewLayer.bounds.size;
```

**Review by janvennemann — APPROVED** [review]
Just a super nit-picky note about variable usage, but nothing dramatic. Approved!

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m (hunk 1 above)
Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m]: Agree. Updated!

**Review by ssjsamir — APPROVED** [review]

**Comment by ssjsamir** [comment]
**FR Passed** Tested with the test case in https://jira.appcelerator.org/browse/TIMOB-27305?focusedCommentId=455767&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-455767, able to see `borderRadius corner` change the border radius control also tested with different values e.g.

```
const view_a = Ti.UI.createView({
	backgroundColor: 'red',
	width: 100,
	height: 50,
	top: 100,
	borderRadius: '20 20px 20 30'
});
const view_b = Ti.UI.createView({
	backgroundColor: 'blue',
	width: 100,
	height: 50,
	top: 150,
	borderRadius: [ '20', '30px', '40dp', 20 ]
});

```
**Test Environment**
```
MacOS Big Sur: 11.0 Beta
Xcode: 12.0 Beta 
Java Version: 1.8.0_242
Android NDK: 21.3.6528147
Node.js: 12.18.1
""NPM":"5.0.0","CLI":"8.0.0""
iphone 8 Sim (14.0 Beta)
```

**Comment by sgtcoolguy** [comment]
manually merged into master

**Comment by Max87ZA** [comment]
Hi guys, I installed Ti.SDK 9.1.0.v20200723064106 today to test it and found a bug - I have views(bar chart) which each bar has borderRadius: [ '8', '0', '8', "0" ], so top left and bottom right corners are rounded. Problem is, that height of the each bar is dynamically generated(based on value from API) and animated and if height is small(let's say 60% of parent), both corners are rounded. But if height is bigger, bottom right corner loose its shape(see attached screenshot, all bars are generated in for loop with same code and there are differences in corner radius). Can you pls look at it and fix it?
Simulator Screen Shot - iPhone 11 Pro - 2020-07-24 at 19 29 18

**Comment by brentonhouse** [comment]
@Max87ZA - Can you post the code used to generate this (or a sample app)?

**Comment by Max87ZA** [comment]
> @Max87ZA - Can you post the code used to generate this (or a sample app)?

Hi Brenton, I also wrote to JIRA ticket [https://jira.appcelerator.org/browse/TIMOB-27305](url), working on standalone app

**Comment by brentonhouse** [comment]
Thanks @Max87ZA!

**Comment by ssaddique** [comment]
Fix verified on build 9.1.0.v20200804082025.

**Test environment**
```
OS Ver:         10.15.3
Xcode Ver:      Xcode 11.6
Appc NPM:       5.0.0
Appc CLI:       8.0.0
Node Ver:       10.17.0
Java Ver:       1.8.0_162
Simulator:      iPhone 5s (11.4), iPhone 8 (12.4), iPhone 11 Pro (13.6)
```

**Pull request closed without merging**